### DISCUSSION TITLE
Printing directly to a connected printer

### DISCUSSION THREAD

**Discussion by christoferd** [NativePHP/laravel#218] [Ideas]
Something that clients sometimes want is a way to print directly to a printer with one button click in a web app.
I have done this with a C# app, but for a web app I have to route it through a native application.

What support does NativePHP have or plan to have for printing?

**Real world use case example:**
Product barcode labels to be printed in a connected special barcode printer. 
This printer may be connected to the Windows machine with a USB cable, or it is connected via Network. Either way it does appear in the list of connected devices within "Printers & scanners" window.

**Comment by osc2nuke** [comment]
Could only find this laravel package for you : https://github.com/charlieuki/receipt-printer

**Reply by christoferd** [reply]
Thank you

**Comment by simonhamp** [comment]
@christoferd printing support is [already there](https://github.com/NativePHP/laravel/blob/main/src/System.php#L29-L58), I think it's potentially just lacking some documentation, which is on my list to tackle over the next couple of weeks.

**Comment by simonhamp** [comment]
[Docs were added](https://nativephp.com/docs/desktop/1/the-basics/system#printing)
